scripts/config-parser/lib/dhcp_config_parser.rb

```diff
@@ -0,0 +1,118 @@
+require "json"
+
+class DhcpConfigParser
+  KEA_CONFIG_FILEPATH = "./data/kea.json".freeze
+  LEGACY_CONFIG_FILEPATH = "./data/export.txt".freeze
+
+  def self.run
+    throw StandardError unless kea_config_exists?
+    throw StandardError unless export_file_exists?
+
+    # Populate these with data from the portal/export before running.
+    # See readme if you're feeling ¯\_(ツ)_/¯
+    shared_network_id = "FITS_####"
+    subnet_list = ["192.168.1.0", "192.168.2.0", "192.168.3.0"]
+
+    compared_reservations = find_missing_reservations(
+      kea_reservations: get_kea_reservations(shared_network_id, File.read(KEA_CONFIG_FILEPATH)),
+      legacy_reservations: get_legacy_reservations(File.read(LEGACY_CONFIG_FILEPATH), subnet_list)
+    )
+  end
+
+  def self.kea_config_exists?
+    File.exist?(KEA_CONFIG_FILEPATH)
+  end
+
+  def self.export_file_exists?
+    File.exist?(LEGACY_CONFIG_FILEPATH)
+  end
+
+  def self.get_kea_reservations(shared_network_id, kea_config)
+    kea_config_hash = JSON.parse(kea_config)
+    shared_networks = kea_config_hash["Dhcp4"]["shared-networks"].select { |shared_network| shared_network["name"].include?(shared_network_id) }
+
+    reservations = shared_networks.inject([]) do |accumulator, shared_network|
+      shared_network["subnet4"].each do |subnet|
+        # accumulator += (subnet["reservations"] || [])
+        accumulator += subnet["reservations"] if subnet["reservations"]
+      end
+
+      accumulator
+    end
+  end
+
+  def self.get_legacy_reservations(export, subnet_list)
+    reservation_fields = ["ip-address", "hw-address", "hostname"]
+    legacy_reservations = []
+
+    subnet_list.each do |subnet|
+      export.scan(/#{subnet.chop}\d{1,3}.(?:[a-fA-F0-9]{12})."[^"]*"."[^"]*"/)
+        .each do |reservation|
```

Comment:
Nitpick/Style: Should align the methods (in my opinion)

```suggestion
      export.scan(/#{subnet.chop}\d{1,3}.(?:[a-fA-F0-9]{12})."[^"]*"."[^"]*"/)
            .each do |reservation|
```

Reply:
😂  Seems like the code formatter disagrees with this! Must remember this for the future (or change the formatter 🙃 )